### PR TITLE
Add a Make target to run all linters.

### PR DESCRIPTION
**Pull request by jotes**
I've added a Make target to run all linters for the python/js code. Also, I've added missing help info about the pyupgrade target.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -77,9 +81,17 @@ black:
...
+pyupgrade-check:
+	"${DC}" run --rm webapp pyupgrade --py38-plus *.py `find pontoon -name \*.py`
+
 types:
 	"${DC}" run --rm -w //app/frontend webapp yarn types
 
+black-check:
+	"${DC}" run --rm webapp black --check pontoon
+
+lint: eslint types flake8 black-check pyupgrade-check
```
Comment (mathjazz) [inline Makefile:93]: Should `check-prettier` be part of the list?

Let's use `check` consistently, either as a prefix of postfix.

File: Makefile [hunk 2]
```diff
@@ -77,9 +81,17 @@ black:
 pyupgrade:
 	"${DC}" run --rm webapp pyupgrade --exit-zero-even-if-changed --py38-plus *.py `find pontoon -name \*.py`
 
+pyupgrade-check:
+	"${DC}" run --rm webapp pyupgrade --py38-plus *.py `find pontoon -name \*.py`
+
 types:
 	"${DC}" run --rm -w //app/frontend webapp yarn types
 
+black-check:
```
Comment (mathjazz) [inline Makefile:90]: Nit, I'd move that next to `black:`.

**Review by mathjazz — COMMENTED** [review]
Thanks, I like that!

Please also add new commands to the `PHONY` list.

**Comment by mathjazz** [comment]
Sorry for not noticing this earlier: Could you please make sure the order of commands is the same in:
- PHONY section
- help section
- the list of commands

**Comment by mathjazz** [comment]
Please rebase.

As mentioned in the [other PR](https://github.com/mozilla/pontoon/pull/1996#pullrequestreview-767788525), please simplify the `ci` command to `test lint` and drop the `server_tests.sh` file.

**Comment by mathjazz** [comment]
As agreed with @jotes, I took over to rebase, made the remaining changes and simplified the `ci` command. I've also changed the order of commands a little and fixed an issue with a broken link.

**Review by eemeli — APPROVED** [review]

**Pull request merged**